core/src/main/java/ai/timefold/solver/core/impl/heuristic/selector/common/iterator/UpcomingSelectionIterator.java

```diff
@@ -76,16 +76,20 @@ public static ElementPosition findUnpinnedDestination(Iterator<ElementPosition>
             ListVariableDescriptor<?> listVariableDescriptor) {
         while (destinationIterator.hasNext()) {
             var destination = destinationIterator.next();
-            if (destination instanceof PositionInList positionInList) {
-                var isPinned = listVariableDescriptor.isElementPinned(null, positionInList.entity(), positionInList.index());
-                if (!isPinned) {
-                    return destination;
-                }
-            } else { // Unassigned element can not be pinned.
+            if (isUnpinned(destination, listVariableDescriptor)) {
                 return destination;
             }
         }
         return null;
     }
 
+    public static boolean isUnpinned(ElementPosition destination, ListVariableDescriptor<?> listVariableDescriptor) {
```

Comment:
Make this `isPinned` - this negation in method name is IMO unnecessary and makes you think about the method more than you should.